edgelet/edgelet-settings/src/base/image.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft. All rights reserved.
+use std::time::Duration;
+
+#[derive(Debug, serde::Deserialize, serde::Serialize, Clone, Default)]
+
+/// This struct is a wrapper for options that allow a user to override the defaults of
+/// the image gabage collection job and customize their settings.
+pub struct ImagePruneSettings {
+    #[serde(with = "humantime_serde")]
+    #[serde(default = "default_cleanup_recurrence")]
+    /// how frequently images should be garbage collected
+    cleanup_recurrence: Duration,
+    #[serde(with = "humantime_serde")]
+    #[serde(default = "default_image_age_cleanup_threshold")]
+    /// minimum (unused) image "age" to be eligible for garbage collection
+    image_age_cleanup_threshold: Duration,
+    /// time in "HH::MM" format when cleanup job runs
+    #[serde(default = "default_cleanup_time")]
+    cleanup_time: String,
+    // is image garbage collection enabled
+    #[serde(default = "default_enabled")]
+    enabled: bool,
+}
+
+impl ImagePruneSettings {
+    pub fn new(
+        cleanup_recurrence: Duration,
+        image_age_cleanup_threshold: Duration,
+        cleanup_time: String,
+        enabled: bool,
+    ) -> ImagePruneSettings {
+        ImagePruneSettings {
+            cleanup_recurrence,
+            image_age_cleanup_threshold,
+            cleanup_time,
+            enabled,
+        }
+    }
+
+    pub fn cleanup_recurrence(&self) -> Duration {
+        self.cleanup_recurrence
+    }
+
+    pub fn image_age_cleanup_threshold(&self) -> Duration {
+        self.image_age_cleanup_threshold
+    }
+
+    pub fn cleanup_time(&self) -> String {
+        self.cleanup_time.clone()
+    }
+
+    pub fn is_enabled(&self) -> bool {
+        self.enabled
+    }
+}
+
```

Comment:
```suggestion

fn hhmm_to_minutes<'de, D>(de: D) -> Result<u32, D::Error>
where
    D: serde::Deserializer<'de>
{
    let value = String::deserialize(de)?;
    let time = chrono::NaiveTime::parse_from_str(&value, "%H:%M")
        .map_err(<D::Error as serde::de::Error>::custom)?;
    Ok(time.hour() * 60 + time.minute())
}

```

Reply:
Add `use chrono::Timelike;` to the prelude to fix the error about `.hour` and `.minute` not being found.

Reply:
Thanks Tarik, this is great!